### DISCUSSION TITLE
How to check authenticated user in client components

### DISCUSSION THREAD

**Discussion by ringge** [saasykits/nextjs-sessionauth-template#63] [Q&A] [answered]
Hello, I tried to use validateRequest() in a client component and got error saying next/headers is only available on server components, so how should we check authenticated user in client components? Anyone has any solutions? Appreciate your help.

**Comment by iamtouha** [comment, marked as answer]
hi @ringge , you can call `validateRequest` in server component only. and then you can receive the user object as prop in the client component.

**Reply by ringge** [reply]
hi @iamtouha thanks so much for your quick reply, I got it working
